src/lib/swapper/swappers/CowSwapper/CowSwapper.test.ts

```diff
@@ -182,18 +179,44 @@ describe('CowSwapper', () => {
       const args = { trade: cowSwapTrade, wallet }
       await swapper.executeTrade(args)
       expect(cowExecuteTrade).toHaveBeenCalledTimes(1)
-      expect(cowExecuteTrade).toHaveBeenCalledWith(COW_SWAPPER_DEPS, args)
+      expect(cowExecuteTrade).toHaveBeenCalledWith(args, COW_SWAPPER_DEPS)
+    })
+
+    it('calls executeTrade on swapper.buildTrade for XDAI', async () => {
+      const cowSwapTrade: CowTrade<KnownChainIds.GnosisMainnet> = {
+        sellAmountBeforeFeesCryptoBaseUnit: '1000000000000000000',
+        buyAmountBeforeFeesCryptoBaseUnit: '14501811818247595090576',
+        sources: [{ name: SwapperName.CowSwap, proportion: '1' }],
+        buyAsset: FOX,
+        sellAsset: XDAI,
```

Comment:
Gnosis trades are same-chain only, a trade of FOX on Ethereum to XDAI on Gnosis is invalid. Might be worth renaming the constants to `FOX_MAINNET` and `FOX_GNOSIS` to avoid confusion

Reply:
@lmyslinski still valid

Reply:
https://github.com/shapeshift/web/pull/4595/commits/f47c41059cfcb95c1005c48c0ad9f6df862c7db2#diff-555038810cdb66f0eacf217a100eec110dbef5a315e5d51a7039e83cc97e245c